### PR TITLE
safe default resources ( and no more assetdatabase asserts)

### PR DESCRIPTION
**Pull request by iTris666**
### Purpose of this PR
Change default resource implementation to make them more reliable as they can be asked at various moments, including while importing after a deleted library because of sanitize.

User can optionaly create an asset with default values per project.

Also include fix to call sanitize OnPreprocessAsset instead of during compilation.

original PR : https://github.cds.internal.unity3d.com/unity/vfx-graph/pull/257
---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.visualeffectgraph/Editor/Models/VFXGraph.cs [hunk 1]
```diff
@@ -18,6 +18,20 @@ namespace UnityEditor.VFX
...
     {
+        void OnPreprocessAsset()
+        {
+            bool isVFX = assetPath.EndsWith(VisualEffectResource.Extension);
+            if (isVFX)
+            {
+                VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
+                if (resource == null)
+                    return;
+                Debug.Log("Sanitizing graph");
```
Comment (julienf-unity) [inline com.unity.visualeffectgraph/Editor/Models/VFXGraph.cs:29]: You forgot to remove this log

**Comment by julienf-unity** [comment]
Added @VladNeykov  as there appear to be a regression in this PR. Described there: https://github.cds.internal.unity3d.com/unity/vfx-graph/pull/257

**Comment by VladNeykov** [comment]
The regression with mesh outputs using shader graph looks fixed.

@iTris666 , can you please take a look at the error message typo mentioned on the last PR as well?
https://github.cds.internal.unity3d.com/unity/vfx-graph/pull/257#issuecomment-19154

**Comment by VladNeykov** [comment]
@iTris666 
ZSyJeJgLh1

The error message also doesn't seem to check how many assets are present, but only if two are created in succession. It's a minor UX issue, not sure if this can be addressed?

**Pull request merged**